toolkit/tools/imagecustomizerapi/bootloader.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerapi
+
+import (
+	"fmt"
+)
+
+type BootLoader struct {
+	Type  BootLoaderType      `yaml:"type"`
+	Reset ResetBootLoaderType `yaml:"reset"`
```

Comment:
Now that we have a bootloader type, I think the `ResetBootLoaderType` is harder to understand.
In the past, it referred to the bootloader configs (specifically kernel args).
Now, for the mic user, does `ResetBootLoaderType`  reset the bootloader type or its configuration?
I think renaming the `ResetBootLoaderType` to `ResetBootLoaderConfigsType` can help the reader...

Reply:
Consequently, `Reset` can be renamed to `ResetConfiguration` for example...

Reply:
Agreed - I've opened a separate part to look into the naming: https://dev.azure.com/mariner-org/ECF/_workitems/edit/10218